File: CS/07_Attachments/PdfAttachmentRelationship.cs
using System;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Attachments;

namespace PdfAttachmentRelationship
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void button1_Click(object sender, EventArgs e)
        {
            string input = @"..\..\..\..\..\..\Data\Attachment.pdf";
            string attachmentPath = @"..\..\..\..\..\..\Data\E-iceblueLogo.png";
            string output = "addAttachments.pdf";
            //Load document from disk
            PdfDocument doc = new PdfDocument();
            doc.LoadFromFile(input);
            //Define PdfAttachment
            PdfAttachment attachment = new PdfAttachment(attachmentPath);
            //Add addachment
            doc.Attachments.Add(attachment, doc, Spire.Pdf.General.PdfAttachmentRelationship.Alternative);
            //Save the document
            doc.SaveToFile(output, FileFormat.PDF);
            //Launch the file
            PDFDocumentViewer(output);
        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}

File: VB/07_Attachments/PdfAttachmentRelationship.vb
Imports Spire.Pdf
Imports Spire.Pdf.Attachments

Namespace PdfAttachmentRelationship
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Specify the input file path for the PDF document
			Dim input As String = "..\..\..\..\..\..\Data\Attachment.pdf"

			' Specify the attachment file path
			Dim attachmentPath As String = "..\..\..\..\..\..\Data\E-iceblueLogo.png"

			' Create a new PdfDocument object
			Dim doc As New PdfDocument()

			' Load the PDF document from the input file path
			doc.LoadFromFile(input)

			' Create a new PdfAttachment object
			Dim attachment As New PdfAttachment(attachmentPath)

			' Add the attachment to the PDF document
			doc.Attachments.Add(attachment, doc, Spire.Pdf.General.PdfAttachmentRelationship.Alternative)

			' Specify the output file name for the PDF document
			Dim output As String = "addAttachments.pdf"

			' Save the modified PDF document to the output file
			doc.SaveToFile(output, FileFormat.PDF)

			' Close the document
			doc.Close()

			' Launch the file
			PDFDocumentViewer(output)
		End Sub
		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace
